pkg/controller/promotionreconciler/prowjobretrigger/reconciler.go

```diff
@@ -0,0 +1,185 @@
+package prowjobretrigger
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/openshift/ci-tools/pkg/rehearse"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/workqueue"
+	prowjobsv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"sigs.k8s.io/controller-runtime"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	cioperatorapi "github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/controller/promotionreconciler/prowjobreconciler"
+	controllerutil "github.com/openshift/ci-tools/pkg/controller/util"
+	"github.com/openshift/ci-tools/pkg/load/agents"
+)
+
+type Options struct {
+	DryRun                bool
+	CIOperatorConfigAgent agents.ConfigAgent
+	ConfigGetter          config.Getter
+	GitHubClient          github.Client
+}
+
+const (
+	ControllerName = "promotion_job_retrigger"
+)
+
+func AddToManager(mgr controllerruntime.Manager, opts Options) error {
```

Comment:
This could be achieved much simpler by simply watching prowjobs in the promotionreconciler and enquing the imagestreamtag(s) they promote to: https://github.com/openshift/ci-tools/blob/2e70752110c84876f70ae7899d491992b527f09c/pkg/controller/promotionreconciler/reconciler.go#L85